nav2_smac_planner/CMakeLists.txt

```diff
@@ -1,153 +1,117 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.10)
 project(nav2_smac_planner)
 
-set(CMAKE_BUILD_TYPE Release) #Debug, Release
+find_package(catkin REQUIRED COMPONENTS
+        roscpp
+        dynamic_reconfigure
+        visualization_msgs
+        mbf_costmap_core
+        nav_msgs
+        nav_msgs
+        geometry_msgs
+        tf2_ros
+        costmap_2d
+        base_local_planner
+        pluginlib
+        angles
+        eigen_conversions
+)
 
-find_package(ament_cmake REQUIRED)
-find_package(nav2_common REQUIRED)
-find_package(rclcpp REQUIRED)
-find_package(rclcpp_action REQUIRED)
-find_package(rclcpp_lifecycle REQUIRED)
-find_package(std_msgs REQUIRED)
-find_package(visualization_msgs REQUIRED)
-find_package(nav2_util REQUIRED)
-find_package(nav2_core REQUIRED)
-find_package(nav2_msgs REQUIRED)
-find_package(nav_msgs REQUIRED)
-find_package(geometry_msgs REQUIRED)
-find_package(builtin_interfaces REQUIRED)
-find_package(tf2_ros REQUIRED)
-find_package(nav2_costmap_2d REQUIRED)
-find_package(pluginlib REQUIRED)
-find_package(eigen3_cmake_module REQUIRED)
-find_package(Eigen3 REQUIRED)
-find_package(angles REQUIRED)
 find_package(ompl REQUIRED)
 
-if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 17)
-endif()
+generate_dynamic_reconfigure_options(
+        cfg/SmacPlannerHybrid.cfg
+        cfg/Smoother.cfg
+)
 
-if(MSVC)
-  add_compile_definitions(_USE_MATH_DEFINES)
-else()
-  add_compile_options(-O3 -Wextra -Wdeprecated -fPIC)
-endif()
+catkin_package(
+        INCLUDE_DIRS include
+        LIBRARIES ${PROJECT_NAME} # TODO ${PROJECT_NAME}_2d ${PROJECT_NAME}_lattice
+        CATKIN_DEPENDS roscpp dynamic_reconfigure visualization_msgs mbf_costmap_core nav_msgs nav_msgs geometry_msgs tf2_ros costmap_2d base_local_planner pluginlib angles
+)
 
 include_directories(
-  include
-  ${OMPL_INCLUDE_DIRS}
+        include
+        ${catkin_INCLUDE_DIRS}
+        ${OMPL_INCLUDE_DIRS}
 )
 
-set(library_name nav2_smac_planner)
-
-set(dependencies
-  rclcpp
-  rclcpp_action
-  rclcpp_lifecycle
-  std_msgs
-  visualization_msgs
-  nav2_util
-  nav2_msgs
-  nav_msgs
-  geometry_msgs
-  builtin_interfaces
-  tf2_ros
-  nav2_costmap_2d
-  nav2_core
-  pluginlib
-  angles
-  eigen3_cmake_module
-)
+set(library_name ${PROJECT_NAME})
 
 # Hybrid plugin
 add_library(${library_name} SHARED
-  src/smac_planner_hybrid.cpp
-  src/a_star.cpp
-  src/collision_checker.cpp
-  src/smoother.cpp
-  src/analytic_expansion.cpp
-  src/node_hybrid.cpp
-  src/node_lattice.cpp
-  src/costmap_downsampler.cpp
-  src/node_2d.cpp
-  src/node_basic.cpp
+        src/smac_planner_hybrid.cpp
+        src/a_star.cpp
+        src/collision_checker.cpp
+        src/smoother.cpp
+        src/analytic_expansion.cpp
+        src/node_hybrid.cpp
+        src/node_lattice.cpp
```

Comment:
smoother is ported, and the other nodes are needed for compiling,, I think because we create somewhere templated versions
analytic_expansion don't remember
but everything compiles anyway